apps/styling/src/app/text.component.ts

```diff
@@ -1,16 +1,34 @@
 /* eslint-disable @angular-eslint/component-selector */
-import { Component, Input } from '@angular/core';
+import { Component } from '@angular/core';
 
 @Component({
   selector: 'text',
   standalone: true,
   template: `
-    <p style="font-size: {{ font }}px; color: {{ color }}">
+    <p>
       <ng-content></ng-content>
     </p>
   `,
+  styles: [
+    `
+      :host(.blue) {
+        color: var(--text-blue);
+      }
+
+      :host(.font-s-15) {
+        font-size: 15px;
+      }
+
+      :host-context(.error) {
+        color: var(--text-error);
+        font-size: 30px;
+      }
+
+      :host-context(.warning) {
+        color: var(--text-warning);
+        font-size: 25px;
+      }
+    `,
+  ],
```

Comment:
error and warning should be inside staticTextComponent. This component should be very generic.

.blue and .font-s-15 are way too specific. If you add new component, you will need to change this component again.

Your css variables are great but make use of them but styling your component from the outside.

Define all properties that can be customize with a default value (you can use a global file as your did, that's fine), but style from the outside.